geopandas/array.py

```diff
@@ -1615,15 +1621,17 @@ def _reduce(self, name, skipna=True, **kwargs):
             f"does not support reduction '{name}'"
         )
 
-    def __array__(self, dtype=None):
+    def __array__(self, dtype=None, copy=None):
         """
         The numpy array interface.
 
         Returns
         -------
         values : numpy array
         """
-        return to_shapely(self)
+        if copy and (dtype is None or dtype == np.dtype("object")):
+            return self._data.copy()
```

Comment:
Numpy doesn't yet actually pass `copy=True` to this function, but my understanding is that it will start doing so in the future, so this code is just to already start honoring the keyword. I am only copying when the resulting dtype will be object, otherwise the subsequent cast will either raise an error or copy anyway